### PR TITLE
auto_init: initialize gnrc_rpl

### PR DESCRIPTION
**Pull request by cgundogan**
Initializes rpl automatically and enables this feature for `gnrc_networking`.

If one interface exists: rpl will be initialized on this interface.
If several interfaces exist, `GNRC_RPL_DEFAULT_NETIF` will be used to initialize rpl.

Initializing rpl on multiple interfaces automatically is not supported for now, but can be achieved from within the application by calling `gnrc_rpl_init()` manually for the desired interfaces.

### CODE REVIEW

File: sys/auto_init/gnrc_rpl/auto_init_gnrc_rpl.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        DEBUG("auto_init_gnrc_rpl: initializing RPL on interface %" PRIkernel_pid "\n",
+              GNRC_RPL_DEFAULT_NETIF);
+        gnrc_rpl_init(GNRC_RPL_DEFAULT_NETIF);
+        return;
+    }
+    DEBUG("auto_init_gnrc_rpl: could not initialize RPL on interface %" PRIkernel_pid" - "
+          "interface does not exist\n", GNRC_RPL_DEFAULT_NETIF);
+    return;
+#endif
+    DEBUG("auto_init_gnrc_rpl: please specify an interface by setting GNRC_RPL_DEFAULT_NETIF\n");
```
Comment (authmillenon) [inline sys/auto_init/gnrc_rpl/auto_init_gnrc_rpl.c]: Put this into `#else` of above statements?

File: sys/auto_init/gnrc_rpl/auto_init_gnrc_rpl.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/*
```
Comment (authmillenon) [inline sys/auto_init/gnrc_rpl/auto_init_gnrc_rpl.c]: Why add a seperate module for that? Can't this function go into `gnrc_rpl`?

**Comment by cgundogan** [comment]
@authmillenon I moved the auto init function to `gnrc_rpl` and created a pseudomodule `auto_init_gnrc_rpl`. This way it is still possible to suppress the auto initialization by not using `auto_init_gnrc_rpl` in the application's `Makefile`.

**Comment by authmillenon** [comment]
> This way it is still possible to suppress the auto initialization by not using auto_init_gnrc_rpl in the application's Makefile.

But why? It makes things needlessly complicated. Something `auto_init ` is supposed to make easier.

**Comment by cgundogan** [comment]
>But why? It makes things needlessly complicated. Something auto_init is supposed to make easier.

An application might not want to initialize RPL at startup and join the next best DODAG, but maybe after a specific amount of time, or as a response to certain event.

**Comment by authmillenon** [comment]
Are there any other opinions on this @emmanuelsearch @OlegHahm @jfischer-phytec-iot?

**Comment by kaspar030** [comment]
+1 for keeping it optional.

**Comment by OlegHahm** [comment]
I agree with @kaspar030 and @cgundogan: sometimes it comes handy for testing not to use RPL.

**Comment by cgundogan** [comment]
The proposed approach auto-inits RPL only when `auto_init_gnrc_rpl` is added to `USEMODULE`. We can also go the other way and activate the auto-initialization by default for RPL and deactivate it with `CFLAGS`. Any opinions?

**Comment by authmillenon** [comment]
I prefer it the way it is at the moment, but maybe adding it to the default modules we, when RPL is present (deactivating it with `DISABLE_MODULE`). Thinking about the optionality of this: why is RPL special in this case and why don't we have something like this for other modules?

**Comment by BytesGalore** [comment]
@cgundogan your current proposed `auto_init_gnrc_rpl` pseudomodule seems to me the a good way to automatically initialize RPL on startup of RIOT.
Usually I do the same calls of the `auto_init_gnrc_rpl` manually in some "startup-function".
I'm against to add this initialization as default and require a developer to actively deactivate it, e.g. by using `DISABLE_MODULE` or a different activation `CFLAG`.
IMO this is a convenience addition introducing a benefit for a specific purpose (e.g. my laziness ;)), not something I would expect as default.

**Comment by emmanuelsearch** [comment]
ACK.

**Comment by emmanuelsearch** [comment]
please squash.

**Comment by cgundogan** [comment]
squashed

**Comment by cgundogan** [comment]
Murdock is happy - go!

**Pull request merged**